aws/table_aws_cloudwatch_metric_data_point.go

```diff
@@ -0,0 +1,280 @@
+		return nil, errors.New("please provide either expression or metric_stat in where clause to use this table")
+	}
+
+	metricDataQueries := types.MetricDataQuery{}
+	metricDataQueries.Id = aws.String(d.EqualsQuals["id"].GetStringValue())
+
+	// Limiting the results
+	maxLimit := int32(100800)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 1 {
+				maxLimit = 1
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	params := &cloudwatch.GetMetricDataInput{
+		MaxDatapoints: aws.Int32(maxLimit),
+	}
+	if d.EqualsQuals["scan_by"] != nil {
+		params.ScanBy = types.ScanBy(d.EqualsQuals["scan_by"].GetStringValue())
+	}
+
+	if d.EqualsQuals["timezone"] != nil {
+		labelOptions := types.LabelOptions{}
+		labelOptions.Timezone = aws.String(d.EqualsQuals["timezone"].GetStringValue())
+		params.LabelOptions = &labelOptions
+	}
+
+	//set the start and end time based on the provided timestamp
+	if d.Quals["timestamp"] != nil {
+		for _, q := range d.Quals["timestamp"].Quals {
+			timestamp := q.Value.GetTimestampValue().AsTime()
+			switch q.Operator {
+			case "=":
+				params.StartTime = aws.Time(timestamp)
+				params.EndTime = aws.Time(timestamp)
+			case ">=", ">":
+				params.StartTime = aws.Time(timestamp)
+			case "<", "<=":
+				params.EndTime = aws.Time(timestamp)
+			}
+		}
+	}
+
+	if params.StartTime == nil {
+		params.StartTime = aws.Time(time.Now().AddDate(0, 0, -1))
+	}
+	if params.EndTime == nil {
+		params.EndTime = aws.Time(time.Now())
+	}
+
+	// set the period based on the duration between the start and end time
+	// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudwatch/types#MetricStat.Period
+	// duration under 15 days - default period will be 60 seconds (1 minute).
+	// duration between 15 and 63 days - default period will be 300 seconds (5 minutes).
+	// duration greater than 63 days ago - default period will be 3600 seconds (1 hour).
+
+	var period int32
+	duration := params.EndTime.Sub(*params.StartTime).Round(time.Hour)
+	if duration.Hours() <= 360 {
+		period = int32(60)
+	} else if duration.Hours() <= 1512 {
+		period = int32(300)
+	} else {
+		period = int32(3600)
+	}
+
+	// override the period if user has provided it in query
+	if d.EqualsQuals["period"] != nil {
+		period = int32(d.EqualsQuals["period"].GetInt64Value())
+	}
+
+	// set Expression or MetricStat
+	if d.EqualsQuals["expression"] != nil {
+		metricDataQueries.Expression = aws.String(d.EqualsQuals["expression"].GetStringValue())
+		metricDataQueries.Period = aws.Int32(period)
+	} else {
+		metric_stat := types.MetricStat{}
+		metric_stat_string := d.EqualsQuals["metric_stat"].GetJsonbValue()
+
+		if metric_stat_string != "" {
+			err := json.Unmarshal([]byte(metric_stat_string), &metric_stat)
+			if err != nil {
+				plugin.Logger(ctx).Error("listCloudWatchMetricDataPoints", "unmarshal_error", err)
```

Comment:
```suggestion
				plugin.Logger(ctx).Error("aws_cloudwatch_metric_data_point.listCloudWatchMetricDataPoints", "unmarshal_error", err)
```
@bigdatasourav could you please make this change wherever we are adding the logging information?

Reply:
updated